src/librustdoc/clean/types.rs

```diff
@@ -57,11 +57,24 @@ crate struct Crate {
     crate primitives: Vec<(DefId, PrimitiveType)>,
     // These are later on moved into `CACHEKEY`, leaving the map empty.
     // Only here so that they can be filtered through the rustdoc passes.
-    crate external_traits: Rc<RefCell<FxHashMap<DefId, (Trait, bool)>>>,
+    crate external_traits: Rc<RefCell<FxHashMap<DefId, TraitWithExtraInfo>>>,
     crate masked_crates: FxHashSet<CrateNum>,
     crate collapsed: bool,
 }
 
+/// This struct is used to wrap additional information added by rustdoc on a `trait` item.
+#[derive(Clone, Debug)]
+crate struct TraitWithExtraInfo {
+    crate trait_: Trait,
+    crate is_spotlight: bool,
+}
+
+impl TraitWithExtraInfo {
+    crate fn new(trait_: Trait, is_spotlight: bool) -> Self {
+        Self { trait_, is_spotlight }
+    }
```

Comment:
Why does this constructor exist? It negates some of the benefit of having a field name and I don't see any reason to have it.

Reply:
Ok, removing it then!